### ISSUE TITLE
fix(security): vulnerabilities found in cactus-example-supply-chain-app 

### ISSUE THREAD

**Issue by zondervancalvez** [hyperledger/cacti#2041]
List of vulnerabilities found in cactus-example-supply-chain-app image during Azure Container scan.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ZONDER~1.CAL/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ZONDER~1.CAL/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

VULNERABILITY ID | PACKAGE NAME | SEVERITY
-- | -- | --
CVE-2021-3807 | ansi-regex | HIGH
CVE-2021-3807 | ansi-regex | HIGH
CVE-2021-43138 | async | HIGH
CVE-2022-22143 | convict | HIGH
CVE-2020-8203 | lodash | HIGH
CVE-2021-23337 | lodash | HIGH
CVE-2022-24771 | node-forge | HIGH
CVE-2022-24772 | node-forge | HIGH
CVE-2021-23358 | underscore | HIGH

</body>

</html>

**Comment by aldousalvarez** [comment]
Hi @petermetz Can you assign me on this one? thank you!

**Comment by aldousalvarez** [comment]
Here are the screenshots of my current issue/blocker
1. Lodash 
![lodash supply chain app](https://user-images.githubusercontent.com/89574062/170949112-823a4e78-4013-4f6d-a972-223cc2f8de3d.JPG)

2. Underscore
![UNDERSCORE Vulnerabilities issue](https://user-images.githubusercontent.com/89574062/170949123-da35936c-db7e-434e-9273-608fe4550e01.JPG)

**Comment by petermetz** [comment]
> Hi @petermetz Can you assign me on this one? thank you!

@aldousalvarez Done, thank you!

**Comment by petermetz** [comment]
> Hello @petermetz While examining each vulnerabilities, both [CVE-2021-23337](https://github.com/advisories/GHSA-35jh-r3h4-6jhm) and [CVE-2021-23358](https://github.com/advisories/GHSA-cf4h-3jhx-xvhq) was already fixed by PR #1816 and #1820. Upon investigation package @hyperledger/cactus-plugin-ledger-connector-besu@1.0.0 at examples/cactus-example-supply-chain-backend is still using the web3-eea on its latest version in NPM https://www.npmjs.com/package/@hyperledger/cactus-plugin-ledger-connector-besu . What is the frequency of update for @hyperledger/cactus-plugin-ledger-connector-besu@1.0.0? Update of the said version from the cactus base code will fix the said vulnerabilities above.
> 
> Here are the screenshots of my current issue/blocker
> 
> 1. Lodash
>    ![lodash supply chain app](https://user-images.githubusercontent.com/89574062/170949112-823a4e78-4013-4f6d-a972-223cc2f8de3d.JPG)
> 2. Underscore
>    ![UNDERSCORE Vulnerabilities issue](https://user-images.githubusercontent.com/89574062/170949123-da35936c-db7e-434e-9273-608fe4550e01.JPG)

@aldousalvarez Oh, that's great, thank you for investigating! We will issue a `1.1` release soon which will then propagate the fixes to the containers as well. I assigned the task to you but at this point it's a no-op except for you having to make sure that this is closed with the issuance of the release `1.1`.
So your responsibilities here are:
1. Wait for the 1.1.0 release and when it happens close this issue stating what happened and why (referring to your description above will help a lot)
2. Mark this issue dependent on that other issue we have for releasing v1.1.0 that I'll create in a second.

**Comment by petermetz** [comment]
Depends on #2054

**Comment by petermetz** [comment]
Also marked this as P3 because the vulnerabilities are on a test or example container that does not get used in production at all.

**Comment by aldousalvarez** [comment]
Hello @petermetz  After examining the remaining vulnerabilities. Below is the table of the proposed solution for the remaining vulnerabilities in the supply-chain-app

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ALDOUS~1.ALV/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ALDOUS~1.ALV/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



VULNERABILITY   ID | PACKAGE NAME | SOLUTION (version) | AFFECTED VERSION | PROPOSED SOLUTION
-- | -- | -- | -- | --
CVE-2021-3807 | ansi-regex | 6.0.1, 5.0.1, 4.1.1, 3.0.1 | Affected versions     >= 6.0.0, < 6.0.1     >= 5.0.0, < 5.0.1     >= 4.0.0, < 4.1.1     >= 3.0.0, < 3.0.1 | already the Solution   (version)   [@hyperledger/cactus-example-supply-chain-backend@1.0.0]
CVE-2021-3807 | ansi-regex | 6.0.1, 5.0.1, 4.1.1, 3.0.1 | Affected versions     >= 6.0.0, < 6.0.1     >= 5.0.0, < 5.0.1     >= 4.0.0, < 4.1.1     >= 3.0.0, < 3.0.1 | (can be fixed by angular supply   chain ticket)
CVE-2021-43138 | async | 3.2.2 / 2.6.4 | Affected versions     >= 3.0.0, < 3.2.2     < 2.6.4 | can be fixed by upgrading the   express-jwt by v6.1.2 on packages/cactus-cmd-api-server/package.json and   fabric-common v.2.2.13 on   packages/cactus-plugin-ledger-connector-fabric/package.json  [@hyperledger/cactus-example-supply-chain-backend@1.0.0]
CVE-2022-22143 | convict | 6.2.3 | < 6.2.3 | already the correct version   based on package.json of cactus-cmd-api-server (need to release   @hyperledger/cactus-cmd-api-server@1.1.0)
CVE-2020-8203 | lodash | 4.17.20 | < 4.17.20 | already done (need to release   @hyperledger/cactus-plugin-ledger-connector-besu@1.1.0 version of the   package)
CVE-2021-23337 | lodash | 4.17.21 | < 4.17.21 | already done (need to release   @hyperledger/cactus-plugin-ledger-connector-besu@1.1.0 version of the   package)
CVE-2022-24771 | node-forge | 1.3.0 | < 1.3.0 | already the correct version   based on package.json of cactus-cmd-api-server (need to release   @hyperledger/cactus-cmd-api-server@1.1.0)
CVE-2022-24772 | node-forge | 1.3.0 | < 1.3.0 | (can be fixed by angular supply   chain ticket)
CVE-2021-23358 | underscore | 1.12.1 | >= 1.3.2, < 1.12.1 | already done (need to release   @hyperledger/cactus-plugin-ledger-connector-besu@1.1.0 version of the   package)
  |   |   |   |  



</body>

</html>

1. ansi-regex
![ansi regex cmd api server](https://user-images.githubusercontent.com/89574062/171606405-44a6b9f5-b608-45f3-8a63-07b2ef76e07c.png)

2. ansi-regex (tested and applied the upgrade angular in supply chain changes)
![ansi-regex after upgrade of angular](https://user-images.githubusercontent.com/89574062/171606492-0ba6378c-3b4a-4d8a-a1f0-b3b076f8da08.jpg)

3. async
a. upgrade on express-jwt 6.1.2
![async after upgrade on express-jwt 6 1 2](https://user-images.githubusercontent.com/89574062/171607188-b45f336c-eea4-466b-aa7d-d01e08ba1090.jpg)


b. upgrade on fabric-common 2.2.13
![async after upgrade on fabric common](https://user-images.githubusercontent.com/89574062/171611222-b9af1372-56e0-4fd6-813f-2fc13b3a14ed.jpg)

4. convict
![image](https://user-images.githubusercontent.com/89574062/171608344-cf1de3ee-623f-4a0d-8df7-e0160ac2f043.png)

5. node-forge
![image](https://user-images.githubusercontent.com/89574062/171608710-d2b367c9-3fa2-4028-b8b4-da6d3c8fc948.png)



6. node-forge (tested and applied the upgrade angular in supply chain changes)
![node forge after upgrade angular supply chain](https://user-images.githubusercontent.com/89574062/171608759-235157f8-ed72-4b10-85e1-9b946fa885e7.jpg)
